docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+    sigstore: https://access.redhat.com/webassets/docker/content/sigstore
+```
+
+If there is no `.yaml` file in the /etc/containers/registries.d/ directory when image-rs want to read it, as the way to obtain policy.json, image-rs should call the GetResource grpc service provided by the attestation-agent. After remote attestation, it is obtained from the user controlled relying KBS through the encrypted channel.
+
+### Image signature verification
+
+After completing the white list check and signature file download, signature verification is divided into the following steps:
+
+1. Parse the signature claim file and separate the JSON payload and cryptographic signature.
+
+2. Read the policy.json's "signedBy" policy, get the public key used to verify the signature from the "keyPath" or "keyData" field.
+
+3. Use the public key to verify the cryptographic signature in the signature claim file, and then compare whether the manifest digest in the JSON payload is consistent with that of the container image and whether the docker reference in the JSON payload is consistent with that of the container image.
+
+### Image decryption
+
+After verifying the container image signature, will start the actual image pulling. When pulling the image layer by layer, if encounter an encrypted image layer, image-rs calls the container image cryptography operation library provided by ocicrypt-rs to decrypt the encrypted image layer as follows:
+
+1. Read the layer's two annotations fields in the image manifest. The `org.opencontainers.image.enc.keys.provider.attestation` field is the annotation packet mentioned above, which contains `PrivateLayerBlockCipherOptions` encrypted by the user's private key. The symmetric key used to encrypt this layer is recorded in this `PrivateLayerBlockCipherOptions`.
+2. ocicrypt-rs calls the keyprovider grpc service provided by the attestation-agent and sends the annotation packet to the attestation-agent.
+3. The attestation-agent performs HW-TEE remote attestation with the relying KBS deployed by the remote user. After passing the attestation, it interacts with KBS through the encrypted channel, and decrypts the `PrivateLayerBlockCipherOptions` using the user's public key in the KBS.
+4. ocicrypt-rs obtains the decrypted `PrivateLayerBlockCipherOptions` from the attestation-agent, reads the symmetric key used to decrypt the container image, and completes the decryption of the container image layer.
+
+## Attestation Agent
+
+The attestation-agent is an indispensable core component in the confidential containers architecture. It undertakes the trust distribution function of the confidential container. In the process of signature verification and decryption of the protected image, the attestation-agent serves as the source of the user's confidential information and the decryptor of the `PrivateLayerBlockCipherOptions`. As mentioned above, it provides two gRPC services for image-rs and ocicrypt-rs to support the above two functions.
```

Comment:
Note the comment I wrote above that you seem to imply that the attestation agent would also be used for encryption. I don't think this is really what you meant. But this highlights that we need to discuss the encryption tools somewhere, and give some command-line examples of how to produce the encrypted images, signatures, etc.